src/http/index.js

```diff
@@ -85,418 +81,26 @@ export default async function http(url, request = {}) {
-   * @param {string} parameterName - Parameter name
-   * @return {object} encoded parameter names and values
-   */
-  if (typeof data === 'string') {
-    try {
-      data = JSON.parse(data);
-      Object.entries(data).forEach(([key, value]) => {
-        if (typeof value === 'object' && !Array.isArray(value)) {
-          data[key] = JSON.stringify(value);
-        }
-      });
-    } catch {
-      return valueEncoder(data, 'reserved');
-    }
-  }
-
-  const encodedQuery = Object.keys(data).reduce((result, parameterName) => {
-    // eslint-disable-next-line no-restricted-syntax
-    for (const [key, value] of formatKeyValue(parameterName, data[parameterName])) {
-      if (value instanceof FileWithData) {
-        result[key] = value.valueOf();
-      } else {
-        result[key] = value;
-      }
-    }
-    return result;
-  }, {});
-
-  return qs.stringify(encodedQuery, { encode: false, indices: false }) || '';
-}
-
-// If the request has a `query` object, merge it into the request.url, and delete the object
-// If file and/or multipart, also create FormData instance
-export function mergeInQueryOrForm(req = {}) {
-  const { url = '', query, form } = req;
-  const joinSearch = (...strs) => {
-    const search = strs.filter((a) => a).join('&'); // Only truthy value
-    return search ? `?${search}` : ''; // Only add '?' if there is a str
-  };
-
-  if (form) {
-    const hasFile = Object.keys(form).some((key) => {
-      const { value } = form[key];
-      return isFile(value) || isArrayOfFile(value);
-    });
-
-    const contentType = req.headers['content-type'] || req.headers['Content-Type'];
-
-    if (hasFile || /multipart\/form-data/i.test(contentType)) {
-      const formdata = buildFormData(req.form);
-      req.formdata = formdata;
-      req.body = formdata;
-    } else {
-      req.body = encodeFormOrQuery(form);
-    }
-
-    delete req.form;
-  }
-
-  if (query) {
-    const [baseUrl, oriSearch] = url.split('?');
-    let newStr = '';
-
-    if (oriSearch) {
-      const oriQuery = qs.parse(oriSearch);
-      const keysToRemove = Object.keys(query);
-      keysToRemove.forEach((key) => delete oriQuery[key]);
-      newStr = qs.stringify(oriQuery, { encode: true });
-    }
-
-    const finalStr = joinSearch(newStr, encodeFormOrQuery(query));
-    req.url = baseUrl + finalStr;
-    delete req.query;
-  }
-  return req;
-}
-
 // Wrap a http function ( there are otherways to do this, consider this deprecated )
-export function makeHttp(httpFn, preFetch, postFetch) {
+function makeHttp(httpFn, preFetch, postFetch) {
   postFetch = postFetch || ((a) => a);
   preFetch = preFetch || ((a) => a);
   return (req) => {
     if (typeof req === 'string') {
       req = { url: req };
     }
-    self.mergeInQueryOrForm(req);
+    mergeInQueryOrForm(req);
     req = preFetch(req);
     return postFetch(httpFn(req));
   };
 }
+
+export {
+  makeHttp,
+  mergeInQueryOrForm,
+  encodeFormOrQuery,
+  isFile,
```

Comment:
We should only export symbols here that actually form the public API of this module. I'd say that only `http` (default export) and `makeHttp` for the publi API of this module.

Other parts of the codebase can look into `http/serializers/request/...`